c2rust-analyze/src/rewrite/expr/convert.rs

```diff
@@ -154,21 +140,6 @@ impl<'tcx> Visitor<'tcx> for ConvertVisitor<'tcx> {
                     Rewrite::Ref(Box::new(self.get_subexpr(ex, 0)), mutbl_from_bool(*mutbl))
                 }
 
-                mir_op::RewriteKind::CastRefToRaw { mutbl } => {
-                    // `addr_of!(*p)` is cleaner than `p as *const _`; we don't know the pointee
-                    // type here, so we can't emit `p as *const T`.
-                    let rw_pl = Rewrite::Deref(Box::new(hir_rw));
-                    Rewrite::AddrOf(Box::new(rw_pl), mutbl_from_bool(*mutbl))
-                }
-                mir_op::RewriteKind::CastRawToRaw { to_mutbl } => {
-                    let method = if *to_mutbl { "cast_mut" } else { "cast_const" };
-                    Rewrite::MethodCall(method.to_string(), Box::new(hir_rw), vec![])
-                }
-                mir_op::RewriteKind::UnsafeCastRawToRef { mutbl } => {
-                    let rw_pl = Rewrite::Deref(Box::new(hir_rw));
-                    Rewrite::Ref(Box::new(rw_pl), mutbl_from_bool(*mutbl))
-                }
-
                 mir_op::RewriteKind::CellNew => {
                     // `x` to `Cell::new(x)`
                     Rewrite::Call("std::cell::Cell::new".to_string(), vec![Rewrite::Identity])
```

Comment:
Why not include `CellNew` in `convert_cast_rewrite`?  It doesn't modify the original expression and is a kind of cast or conversion, similar to `CellFromMut`, which is included in `convert_cast_rewrite`?

Reply:
That makes sense - I've moved it.